Sources/OpenCombine/Publishers/Publishers.Encode.swift

```diff
@@ -0,0 +1,115 @@
+        public typealias Failure = Error
+
+        /// The kind of values published by this publisher.
+        public typealias Output = Coder.Output
+
+        public let upstream: Upstream
+
+        private let encoder: Coder
+
+        public init(upstream: Upstream, encoder: Coder) {
+            self.upstream = upstream
+            self.encoder = encoder
+        }
+
+        /// This function is called to attach the specified `Subscriber`
+        /// to this `Publisher` by `subscribe(_:)`
+        ///
+        /// - SeeAlso: `subscribe(_:)`
+        /// - Parameters:
+        ///     - subscriber: The subscriber to attach to this `Publisher`.
+        ///                   once attached it can begin to receive values.
+        public func receive<Receiver: Subscriber>(subscriber: Receiver)
+            where Failure == Receiver.Failure, Output == Receiver.Input {
+                let encodeSubscriber = _Encode<Upstream, Receiver, Coder>(
+                    downstream: subscriber,
+                    encoder: encoder
+                )
+                upstream.receive(subscriber: encodeSubscriber)
+        }
+    }
+}
+
+private final class _Encode<Upstream: Publisher,
+                            Downstream: Subscriber,
+                            Coder: TopLevelEncoder>
+    : OperatorSubscription<Downstream>,
+      Subscriber,
+      CustomStringConvertible,
+      Subscription
+    where Coder.Output == Downstream.Input,
+          Upstream.Output: Encodable,
+          Downstream.Failure == Error {
+
+    typealias Input = Upstream.Output
+    typealias Failure = Upstream.Failure
+    typealias Output = Downstream.Input
+
+    private let _encoder: Coder
+    private var _demand: Subscribers.Demand = .none
+
+    var description: String { return "Encode" }
+
+    init(downstream: Downstream, encoder: Coder) {
+        self._encoder = encoder
+        super.init(downstream: downstream)
+    }
+
+    func receive(subscription: Subscription) {
+        upstreamSubscription = subscription
+        subscription.request(.unlimited)
+        downstream.receive(subscription: self)
+    }
+
+    func receive(_ input: Input) -> Subscribers.Demand {
+        do {
+            let value = try _encoder.encode(input)
+            return downstream.receive(value)
+        } catch {
+            downstream.receive(completion: .failure(error))
+            cancel()
+            return .none
+        }
+    }
+
+    func receive(completion: Subscribers.Completion<Failure>) {
+        switch completion {
+        case .finished:
+            downstream.receive(completion: .finished)
+        case .failure(let error):
+            downstream.receive(completion: .failure(error))
+        }
+    }
+
+    func request(_ demand: Subscribers.Demand) {
+        _demand = demand
+    }
+}
+
+extension Publisher {
+    public func encode<Coder>(
+        encoder: Coder
+    ) -> Publishers.Encode<Self, Coder>
+        where Coder : TopLevelEncoder
```

Comment:
No space before colon

Reply:
Sorry about that, I didn't mess with the indentation and they're a little wonky in the Publisher.swift file.

Reply:
It's okay, I understand. Those commented out declarations in Publisher.swift are taken from the generated interface from Combine, that's the style they use in generated interfaces.